roles/translator/src/proxy/bridge.rs

```diff
@@ -95,49 +137,74 @@ impl Bridge {
         let tx_sv2_submit_shares_ext = self_
             .safe_lock(|s| s.tx_sv2_submit_shares_ext.clone())
             .unwrap();
+        let target = self_.safe_lock(|s| s.target.clone()).unwrap();
         task::spawn(async move {
             loop {
-                let (sv1_submit, extranonce) = rx_sv1_submit.clone().recv().await.unwrap();
+                let upstream_target: [u8; 32] =
+                    target.safe_lock(|t| t.clone()).unwrap().try_into().unwrap();
+                let mut upstream_target: Target = upstream_target.into();
+                self_
+                    .safe_lock(|s| s.channel_factory.set_target(&mut upstream_target))
+                    .unwrap();
+
+                let (sv1_submit, extrnonce) = rx_sv1_submit.clone().recv().await.unwrap();
                 let channel_sequence_id =
                     self_.safe_lock(|s| s.channel_sequence_id.next()).unwrap() - 1;
-                let sv2_submit: SubmitSharesExtended =
-                    Self::translate_submit(&self_, channel_sequence_id, sv1_submit, &extranonce)
-                        .unwrap();
-                tx_sv2_submit_shares_ext.send(sv2_submit).await.unwrap();
+                let sv2_submit: SubmitSharesExtended = self_
+                    .safe_lock(|s| {
+                        s.translate_submit(channel_sequence_id, sv1_submit, extrnonce)
+                            .unwrap()
```

Comment:
Potential poison lock

Reply:
I received a poison error while running a local sim on this branch. Not sure where the origin of the error was though

Reply:
yep better to get the result out of the lock  and then unwrap there